demos/GMW_demo/gmw_and.js

```diff
@@ -0,0 +1,174 @@
+          jiff.deferreds[opId][sendId] = new jiff.helpers.Deferred();
+        }
+        jiff.deferreds[opId][sendId].resolve(opts);
+      });
+    }
+    // compute ci to return
+    Promise.all(my_count).then(function (v) {
+      //v= [ [ 0 ], [ 0, 0, 0, 0 ], [ 0, 1, 0, 1 ] ]
+
+      // choose one result from 4 options
+      var find = function (msgls,csecret) {
+        var choose=2*csecret[0]+csecret[1];// 0,1,2,3
+        var one=msgls[choose];
+        return one;
+      }
+
+      var re=0;
+      for ( i=0;i<v.length;i++) {
+        var cur=v[i];
+        if (cur.length===1) {
+          re=re^cur[0];
+        } else {
+          var ans=find(cur,csecret);// ot select one answer from 4 options
+          re=re^ans;
+        }
+      }
+
+      var _remaining=jiff.party_count-jiff.id+1;
+      for (i =jiff.id;i<=jiff.party_count;i++) {
+        if (jiff.deferreds[share_id][i]!=='undefined') {
+          var mpromise=jiff.deferreds[share_id][i].promise;
+          (function (mpromise, i) { // p_id is modified in a for loop, must do this to avoid scoping issues.
+            mpromise.then(function ()  {
+              delete jiff.deferreds[share_id][i];
+              _remaining--;
+              if (_remaining === 0) {
+                delete jiff.deferreds[share_id];
+              }
+            });
+          })(mpromise, i);
+        }
+      }
+      final_deferred.resolve(re);
+
+    });
+
+    return final_promise;
+  };
+
+  return new share1.jiff.SecretShare(share1.when_both_ready(share2, ready), share1.holders, Math.max(share1.threshold, share2.threshold), share1.Zp);
+
+}
+
+/**
+ * compute 4 OT options in a list of length 4
+ * @function OTGate
+ * @param {number} list of secrect share(length 2)
+ * @returns {object} list of number(length=4)
+ */
+function OTGate(i_shares) {
+  var OT_option = function (cx,cy,i_shares) {
+    var op=(cx&i_shares[1])^(cy&i_shares[0]);
+    return op;
+  };
+  var opt1=OT_option(0,0,i_shares);
+  var opt2=OT_option(0,1,i_shares);
+  var opt3=OT_option(1,0,i_shares);
+  var opt4=OT_option(1,1,i_shares);
+  var re=[];
+  re.push(opt1);
+  re.push(opt2);
+  re.push(opt3);
+  re.push(opt4);
+  return re;
+}
+
+
+/**
+ * function to get promises list to be resolve.
+ * @function initial_deferreds
+ * @param {module:jiff-client~JIFFClient} jiff - the jiff instance
+ * @param {number} share_id of the deferreds
+ * @returns {object} list of promise tp be resolve
+ */
+function initial_deferreds(jiff,share_id) {
+  if (jiff.deferreds[share_id] == null) {
+    jiff.deferreds[share_id] = {};
+  }
+  var re=[];// list of promises to resolve
+  for (var i=jiff.id;i<=jiff.party_count;i++) {
+    if (jiff.deferreds[share_id][i] == null) { // not ready, setup a deferred
+      jiff.deferreds[share_id][i] = new jiff.helpers.Deferred();
+    }
+    re.push( jiff.deferreds[share_id][i].promise);
+
+  }
+  return re;
+}
+
+module.exports = {
```

Comment:
this looks good. only gmw_and is exported to whichever files require this, which is correct.